.github/workflows/build-and-test.yml

```diff
@@ -13,15 +12,17 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [18.x, 20.x, 21.x]
-
+        node-version: [20.x, 22.x, 23.x]
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+    - run: corepack enable
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v3
+      uses: actions/setup-node@v4
       with:
         node-version: ${{ matrix.node-version }}
         cache: 'yarn'
-    - run: yarn install --frozen-lockfile
+    - run: corepack enable
+    - run: yarn install --immutable
     - run: yarn build
     - run: yarn test
+    - run: yarn dlx @arethetypeswrong/cli . --pack
```

Comment:
this should either be pinned to a specific sha or we should just add it as a `devDependency` imo